pg-extend/src/pg_datum.rs

```diff
@@ -110,6 +110,43 @@ impl From<i64> for PgDatum {
     }
 }
 
+impl TryFromPgDatum for String {
```

Comment:
Just asking for the sake of discussion, but do we want to provide conversions for CStr/CString instead, and allow users to decide to take the hit on the cost of the conversion, rather than this library doing it?

I think this is easier to use, but people may want to have a lower cost abstraction? I don't know, maybe we want both?

Reply:
I like using `String` because it simplifies things for people like me who've never done this before. That said, you make a very good point. I'll take a stab at adding conversions for `CStr`/`CString`  as well :thumbsup:

Reply:
I've added an impl for `CString` here as well. I don't think I can do CStr because it's not Sized, or if there is I don't know how 🤷‍♂️

Reply:
Yup. Cstr would have only be possible if the underlying system didn’t allocate, but it does. So CString is the only option I think.